index.js

```diff
@@ -0,0 +1,114 @@
+import http from 'http';
+import fs from 'fs';
+import url from 'url'
+
+
+const requestHandler = (req, res) => {
+    const { headers, method } = req;
+    const parsedUrl = url.parse(req.url, true);
+    let body = [];
+
+    req.on('error', (err) => {
+        console.error(err.stack);
+    })
+
+
+
+    if (method === "GET" && req.url === "/") {
+        res.statusCode = 200;
+        res.setHeader('Content-Type', 'text/html');
+        res.write('<html>');
+        res.write('<head><title>Enter Message</title><head>');
+        res.write('<body><form action="/create-message" method="POST"><input type="text" name="message"><button type="submit">Submit</button></form></body>')
+        res.write('</html>');
+        return res.end();
+    }
+
+
+    if (method === "GET" && req.url.match(/^\/\?name=([^\/]+)/)) {
+        const name = parsedUrl.query.name;
+        res.write(`Hello, ${name}`)
+        return res.end();
+    }
+
+    if (method === "GET" && req.url === "/messages") {
+        res.statusCode = 200;
+        res.setHeader('Content-Type', 'text/html');
+        res.write('<html>');
+        res.write('<head><title>Messages</title><head>');
+        res.write('<body>')
+        res.write('<ul>')
+        res.write('<li>1 ---> Yatin<li><li>2 ---> Arpit<li>')
+        res.write('<li>3 ---> Gautam<li><li>4 ---> Kritika<li>')
+        res.write('</ul>')
+        res.write('</body>')
+        res.write('</html>');
+        return res.end();
+
+    }
+
+    const regex = new RegExp('^/users/([^/]+)/messages/([^/]+)$');
```

Comment:
If the expression is not changing at runtime better to directly use regex without RegExp